src/app/heroes/heroes.component.spec.ts

```diff
@@ -37,4 +50,13 @@ describe('Heroes Component', ()=> {
     expect(heroesCmp.heroes[1]).toBe(HEROES[2]);
   })
 
+  it('should call delelteHero with the correct hero', ()=> {
+    heroesCmp.heroes = HEROES;
+    mockHeroSvc.deleteHero.and.returnValue(of(true));
+
+    heroesCmp.delete(HEROES[1]);
+
+    expect(mockHeroSvc.deleteHero).toHaveBeenCalledWith(HEROES[1])
```

Comment:
could you try sth for me @asem1789 ?

add this test again and change this line to the following:
`expect((mockHeroSvc as HeroService).deleteHero).toHaveBeenCalledWith(HEROES[1])`

See if it passes this way and let me know!

Reply:
hi,
if I put a type for mockHeroSvc
```
let mockHeroSvc: any ;
```

both of ways works,

```
  expect((mockHeroSvc as HeroService).deleteHero).toHaveBeenCalledWith(HEROES[1])
   expect(mockHeroSvc.deleteHero).toHaveBeenCalledWith(HEROES[1])
```

Reply:
![pass](https://user-images.githubusercontent.com/40121654/110470634-c1a9e300-80e3-11eb-9003-065e842b8a42.PNG)

Reply:
Great! But i am still confused about this. i will play with your tests later and see what's going on.